spyrmsd/rmsd.py

```diff
@@ -373,3 +378,189 @@ def rmsdwrapper(
+    -------
+    List[float]
+        RMSDs
+
+    Notes
+    -----
+    Timeout implementation inspired by https://superfastpython.com/task-with-timeout-child-process/
+    """
+
+    if not isinstance(mols, list):
+        mols = [mols]
+
+    # RMSD is computed by the child process
+    # The results need to be shared with the parent process
+    # https://docs.python.org/3/library/multiprocessing.html#sharing-state-between-processes
+    result = Value("f")
+    process = Process(
+        target=_rmsd_process,
+        args=(molref, mols, result, symmetry, center, minimize, strip, cache),
+    )
+
+    process.start()
+    process.join(timeout=timeout)
+
+    # Check if the process finished running successfully
+    if process.exitcode != 0:
+        # Actually terminate the process if it still running
+        if process.is_alive():
+            process.terminate()
+
+        return [np.nan] * len(mols)
+    else:
+        # Retrieve the result from the finished job.
+        # Currently MyPy gives an error, it's being worked on: https://github.com/python/typeshed/issues/8799
+        return result.value  # type: ignore[attr-defined]
+
+
+def prmsdwrapper(
+    molrefs: Union[molecule.Molecule, List[molecule.Molecule]],
+    mols: Union[molecule.Molecule, List[molecule.Molecule]],
+    num_workers: Union[int, None] = 1,
+    symmetry: bool = True,
+    center: bool = False,
+    minimize: bool = False,
+    strip: bool = True,
+    cache: bool = True,
+    timeout: Optional[float] = None,
+) -> Any:
+    """
+    Compute RMSD between two molecules with a timeout.
+
+    Parameters
+    ----------
+    molrefs: Union[molecule.Molecule, List[molecule.Molecule]]
+        Reference molecule
+    mols: Union[molecule.Molecule, List[molecule.Molecule]]
+        Molecules to compare to reference molecule
+    num_workers: int
+        Amount of processor to use for the parallel calculations
+    symmetry: bool, optional
+        Symmetry-corrected RMSD (using graph isomorphism)
+    center: bool, optional
+        Center molecules at origin
+    minimize: bool, optional
+        Minimised RMSD (using the quaternion polynomial method)
+    strip: bool, optional
+        Strip hydrogen atoms
+    timeout: float, optional
+        After how many seconds to stop the RMSD calculations
+
+    Returns
+    -------
+    List[float]
+        RMSDs
+    """
+
+    # Ensure the num_workers is less or equal than the max number of CPUs.
+    # MyPy doesn't like the min() operator since os.cpu_count() can return None in some cases
+    num_workers = min(num_workers, os.cpu_count()) if os.cpu_count() is not None else 1  # type: ignore[type-var]
+
+    # Cast the molecules to lists if they aren't already
+    if not isinstance(molrefs, list):
+        molrefs = [molrefs]
+    if not isinstance(mols, list):
+        mols = [mols]
+
+    # Match the length of the molref
+    if len(molrefs) == 1 and len(molrefs) < len(mols):
+        molrefs = molrefs * len(mols)
```

Comment:
The current behaviour is like this: If a user inputs 1 molref and a list of mols (`[mol1, mol2, ...]` to compare it to (so this could be different docked poses), what happens is that multiple `rmsdwrapper` calculations are performed across different cores like this: 
```python
rmsdwrapper(molref, mol1) #core 1
rmsdwrapper(molref, mol2) #core 2
...
```
So this is just using an embarrassingly parallel approach using the "regular" `rmsdwrapper`, the graph isomorphisms calculations aren't even really being parallelized explicitly. Thinking more about it, I'm honestly not even sure how this interacts with the graph caching either (because of the multicore aspect). 
My implementation had mostly my own use-case in mind, which I thought would generalize decently well. Basically I wanted to accelerate the workload of calculatng the RMSD of a few ligandposes from different docking methods after a virtual screening run. However, considering your insights, maybe my approach was/is too simplistic and needs to be improved. However I think this would introduce even more complexity, so we will need to think things through carefully (so basically continue at it like we have been since the start haha).